### PR TITLE
switch to operator v1

### PR DESCRIPTION
**Pull request by deads2k**
/assign @sanchezl

### CODE REVIEW

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation/images"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+)
+
+type ConfigObserver struct {
+	*configobserver.ConfigObserver
+}
+
+//NewConfigObserver initializes a new configuration observer.
+func NewConfigObserver(
```
Comment (mfojtik) [inline pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: nit: space

File: pkg/operator/configobservation/etcd/observe_etcd.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+)
+
+// observeStorageURLs observes the storage config URLs. If there is a problem observing the current storage config URLs,
+// then the previously observed storage config URLs will be re-used.
+func ObserveEtcdEndpoints(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```
Comment (mfojtik) [inline pkg/operator/configobservation/etcd/observe_etcd.go]: nit: match func name

File: pkg/operator/configobservation/etcd/observe_etcd.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		errs = append(errs, fmt.Errorf("endpoints/etcd.kube-system: alpha.installer.openshift.io/dns-suffix annotation not found"))
+		return previouslyObservedConfig, errs
+	}
+	for subsetIndex, subset := range etcdEndpoints.Subsets {
+		for addressIndex, address := range subset.Addresses {
+			if address.Hostname == "" {
+				errs = append(errs, fmt.Errorf("endpoints/etcd.kube-system: subsets[%v]addresses[%v].hostname not found", subsetIndex, addressIndex))
+				continue
+			}
+			storageURLs = append(storageURLs, "https://"+address.Hostname+"."+dnsSuffix+":2379")
```
Comment (mfojtik) [inline pkg/operator/configobservation/etcd/observe_etcd.go]: we hardcode ports?

File: pkg/operator/configobservation/images/observe_images.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+)
+
+func ObserveInternalRegistryHostname(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+
+	internalRegistryHostnamePath := []string{"imagePolicyConfig", "internalRegistryHostname"}
+	if currentInternalRegistryHostname, _, _ := unstructured.NestedString(existingConfig, internalRegistryHostnamePath...); len(currentInternalRegistryHostname) > 0 {
```
Comment (mfojtik) [inline pkg/operator/configobservation/images/observe_images.go]: why ignoring error?

File: pkg/operator/workload_controller.go [hunk 5]
```diff
@@ -90,70 +91,23 @@ func (c OpenShiftAPIServerOperator) sync() error {
...
+	case operatorsv1.Unmanaged:
 		return nil
 
-	case operatorsv1alpha1.Removed:
+	case operatorsv1.Removed:
 		// TODO probably need to watch until the NS is really gone
 		if err := c.kubeClient.CoreV1().Namespaces().Delete(targetNamespaceName, nil); err != nil && !apierrors.IsNotFound(err) {
 			return err
 		}
-		operatorConfig.Status.TaskSummary = "Remove"
```
Comment (mfojtik) [inline pkg/operator/workload_controller.go]: how do we report removing?

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 6]
```diff
@@ -3,6 +3,9 @@ package operator
 import (
 	"fmt"
 
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
```
Comment (mfojtik) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: nit: move down

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 7]
```diff
@@ -23,18 +26,13 @@ import (
...
-	}
-
+func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, originalOperatorConfig *v1alpha1.OpenShiftAPIServerOperatorConfig) (bool, error) {
 	errors := []error{}
 	var err error
+	operatorConfig := originalOperatorConfig.DeepCopy()
 
 	directResourceResults := resourceapply.ApplyDirectly(c.kubeClient, v311_00_assets.Asset,
 		"v3.11.0/openshift-apiserver/ns.yaml",
-		"v3.11.0/openshift-apiserver/public-info-role.yaml",
```
Comment (mfojtik) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: we don't need these?

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 8]
```diff
@@ -127,30 +146,22 @@ func manageOpenShiftAPIServerClientCA_v311_00_to_latest(client coreclientv1.Core
...
-	configMap := resourceread.ReadConfigMapV1OrDie(v311_00_assets.MustAsset("v3.11.0/openshift-apiserver/public-info.yaml"))
-	if operatorConfig.Status.CurrentAvailability != nil {
-		configMap.Data["version"] = operatorConfig.Status.CurrentAvailability.Version
-	} else {
-		configMap.Data["version"] = ""
+	required.Spec.Template.Spec.Containers[0].ImagePullPolicy = corev1.PullAlways
+	if len(imagePullSpec) > 0 {
+		required.Spec.Template.Spec.Containers[0].Image = imagePullSpec
 	}
+	required.Spec.Template.Spec.Containers[0].Args = append(required.Spec.Template.Spec.Containers[0].Args, fmt.Sprintf("-v=%d", 2))
```
Comment (mfojtik) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: @deads2k how can one raise the loglevel? will operator stomp any changes to args?

**Comment by deads2k** [comment]
```
Error: Error applying plan:

4 error(s) occurred:

* module.masters.aws_lb_target_group_attachment.master[6]: index 2 out of range for list aws_instance.master.*.private_ip (max 2) in:

${aws_instance.master.*.private_ip[count.index / var.target_group_arns_length]}
* aws_route53_record.etcd_a_nodes[2]: index 2 out of range for list module.masters.ip_addresses (max 2) in:

${module.masters.ip_addresses[count.index]}
* module.masters.aws_lb_target_group_attachment.master[7]: index 2 out of range for list aws_instance.master.*.private_ip (max 2) in:

${aws_instance.master.*.private_ip[count.index / var.target_group_arns_length]}
* module.masters.aws_lb_target_group_attachment.master[8]: index 2 out of range for list aws_instance.master.*.private_ip (max 2) in:

${aws_instance.master.*.private_ip[count.index / var.target_group_arns_length]}

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


level=fatal msg="Error executing openshift-install: failed to fetch Cluster: failed to generate asset \"Cluster\": failed to run terraform: failed to execute Terraform: exit status 1"
2018/11/20 16:04:48 Container setup in pod e2e-aws failed, exit code 1, reason Error
Another process exited
2018/11/20 16:04:58 Container test in pod e2e-aws failed, exit code 1, reason Error
2018/11/20 16:07:53 Container teardown in pod e2e-aws completed successfully
2018/11/20 16:07:53 Copying artifacts from e2e-aws into /logs/artifacts/e2e-aws
2018/11/20 16:07:54 Copied 1.11Mi of artifacts from e2e-aws to /logs/artifacts/e2e-aws
2018/11/20 16:07:59 Ran for 14m25s
error: could not run steps: could not wait for pod to complete: could not wait for pod completion: the pod ci-op-q4tw8w1b/e2e-aws failed after 7m45s (failed containers: setup, test):  unknown

Container setup exited with code 1, reason Error
```

**Pull request merged**